### ISSUE TITLE
`ModuleDict` should implement all `dict` API

### ISSUE THREAD

**Issue by bryant1410** [pytorch/pytorch#70095]
### 🚀 The feature, motivation and pitch

Given that `ModuleDict` is a dictionary that holds modules, it should implement its API. A user should manipulate it as a `dict`, not having to think which operations are actually implemented and which don't (it's like re-inventing the wheel in a way).

Methods not implemented, that I have identified (see [`dict` reference](https://docs.python.org/3/library/stdtypes.html#dict)):

* `copy()`
* `get(key[, default])`
* `pop` doesn't support a default.
* `popitem()`
* `reversed(d)`
* `setdefault(key[, default])`
* `d | other`
* `d |= other`

Maybe the simpler way to do it is by inheriting from `dict`? Not sure about the consequences of multiple inheritances here.

### Alternatives

One alternative is to do nothing at all. However, the user needs to know which methods are implemented and how, instead of relying on their knowledge on how a dictionary usually works.

Can't come up with others.

**Comment by albanD** [comment]
This is a duplicate of https://github.com/pytorch/pytorch/issues/68476 ?

**Comment by bryant1410** [comment]
> This is a duplicate of #68476 ?

It's a superset of it.

BTW, I searched for something similar before posting!! Not to blame somebody else but I always end up concluding that GitHub search is suboptimal :sweat_smile:

**Comment by jbschlosser** [comment]
Should be addressed by #69403 once it lands (it's landing now :) )

**Comment by bryant1410** [comment]
Oh, great timing!! Hahaha
